src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -171,9 +172,13 @@ else if (exercise instanceof QuizExercise || exercise instanceof ModelingExercis
                 // in case the participation was finished before, we set it to initialized again so that the user sees the correct button "Open modeling editor" on the client side
                 participation.setInitializationState(INITIALIZED);
             }
+
             if (!Optional.ofNullable(participation.getInitializationDate()).isPresent()) {
                 participation.setInitializationDate(ZonedDateTime.now());
             }
+
+            // initialize a modeling or text submission (depending on the exercise type), it will not do anything in the case of a quiz exercise
+            initializeSubmission(participation, exercise);
```

Comment:
While the participation will only be created if there is no participation yet, the submission could be created multiple times (there is no check if a submission already exists). Or am I seeing this wrong?
I think it's strange that we are handling participations and submissions differently here.

Reply:
You are right, I added a check there.